src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MemberNameCheck.java

```diff
@@ -150,12 +150,16 @@ public MemberNameCheck() {
 
     @Override
     protected final boolean mustCheckName(DetailAST ast) {
-        final DetailAST modifiersAST =
-            ast.findFirstToken(TokenTypes.MODIFIERS);
-        final boolean isStatic = modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
+        final DetailAST modifiersAST = ast.findFirstToken(TokenTypes.MODIFIERS);
+        boolean isStatic = false;
```

Comment:
I wonder about simply adding a `modifiersAST != null` clause
```java
boolean isStatic = modifiersAST != null &&
    modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
```
and then leaving the rest of the method unchanged/not necessarily introducing `Scope.isInPatternDefinition`

Would that work?

Reply:
>I wonder about simply adding a modifiersAST != null clause

This would work. However, since we are checking instance variables in this check, I thought it was best not to obfuscate the intention of the additional code.  I wanted to make it clear that since this `VARIABLE_DEF` AST is in a pattern definition, it has no modifiers and is also not an instance variable. 

[Check documentation](https://checkstyle.sourceforge.io/config_naming.html#MemberName) for quick reference.

Reply:
I like your thinking there, but I think in general it's good practice whenever a variable may possibly be null to check for that before accessing it and avoid a NPE, and it should be more future proof in this case to go ahead and be explicit (not relying on "if not in pattern definition, `modifiersAST != null`"). For example, maybe `MODIFIERS` could also be null outside the context of a pattern as a result of some other change.

Reply:
Add new code added has to pass pitest, which means there must be a test to prove the null check is needed. I don't see how this is possible unless you have an input that will produce a NPE.

Reply:
>  I don't see how this is possible unless you have an input that will produce a NPE.

The modifiers AST is null in this case, so that's not a problem.